### ISSUE TITLE
Add more unit tests for continuous integration

### ISSUE THREAD

**Issue by daniel-luberda** [luberda-molinet/FFImageLoading#861]
- [ ] Add more Mock platform tests
  - [ ] DownloadCache tests
  - [ ] Better ImageService tests coverage
- [ ] Add Android Platform specific tests
  - [ ] MemoryCache tests
  - [ ] DiskCache tests
- [ ] Add iOS Platform specific tests
  - [ ] MemoryCache tests
  - [ ] DiskCache tests
- [ ] Add Mac Platform specific tests
  - [ ] MemoryCache tests
  - [ ] DiskCache tests
- [ ] Add Windows Platform specific tests
  - [ ] MemoryCache tests
  - [ ] DiskCache tests
- [ ] Add Tizen Platform specific tests
  - [ ] MemoryCache tests
  - [ ] DiskCache tests

https://github.com/luberda-molinet/FFImageLoading/tree/master/source/Tests/FFImageLoading.Tests

**Comment by jessejiang0214** [comment]
I met the same problem with FFImageLoading.Transformations, is there any quick way to get rid of this?
Like inject different instance etc?

**Comment by daniel-luberda** [comment]
Could you describe what the problem is? Is it related to unit tests? Usually deleting `bin`/`obj` packages and reopening solution should fix a lot of VS issues

**Comment by jessejiang0214** [comment]
Yes, we do the unit test, we use ```Xamarin.Forms.Mocks``` to mock Xamarin.Forms. In our XF project, we use transformation like this.

```
                <ffimageloading:CachedImage.Transformations>
                    <fftransformations:CircleTransformation BorderSize="0" />
                </ffimageloading:CachedImage.Transformations>
```

And it throws this exception ```FFImageLoading.Transformations.Common.DoNotReferenceMessage```

I used 2.4.3.840

**Comment by daniel-luberda** [comment]
@jessejiang0214 Hi. It should be fixed with: https://github.com/luberda-molinet/FFImageLoading/commit/f3c2cb8882738101a730dfd3985908fad8a4224d

Also please make sure you follow this:
https://github.com/luberda-molinet/FFImageLoading/wiki/Unit-Testing----how-to-enable

**Comment by jessejiang0214** [comment]
Hi @daniel-luberda , cool, is there any nuget contain that fix?

**Comment by jessejiang0214** [comment]
Hi @daniel-luberda I tested 2.4.10.966-pre in pipline, it fixed this issue, thanks.
